sql/migrations/warehouse/000001_create_tables.up.sql

```diff
@@ -0,0 +1,139 @@
+
+--
+-- wh_staging_files
+--
+
+DO $$ BEGIN
+    DROP TYPE wh_staging_state_type;
+    EXCEPTION WHEN OTHERS THEN null;
+END $$;
+
+CREATE TABLE IF NOT EXISTS wh_staging_files (
+    id BIGSERIAL PRIMARY KEY,
+    location TEXT NOT NULL,
+    source_id VARCHAR(64) NOT NULL,
+    destination_id VARCHAR(64) NOT NULL,
+    schema JSONB NOT NULL,
+    error TEXT,
+    status VARCHAR(255),
+    first_event_at TIMESTAMP,
+    last_event_at TIMESTAMP,
+    total_events BIGINT,
+    created_at TIMESTAMP NOT NULL,
+    updated_at TIMESTAMP NOT NULL);
+
+ALTER TABLE wh_staging_files
+    ADD COLUMN IF NOT EXISTS first_event_at TIMESTAMP,
+    ADD COLUMN IF NOT EXISTS last_event_at TIMESTAMP,
+    ADD COLUMN IF NOT EXISTS total_events BIGINT;
+
+CREATE INDEX IF NOT EXISTS wh_staging_files_id_index ON wh_staging_files (source_id, destination_id);
+
+ALTER TABLE wh_staging_files ALTER COLUMN status TYPE VARCHAR(255);
+
+--
+-- wh_load_files
+--
+
+CREATE TABLE IF NOT EXISTS wh_load_files (
+    id BIGSERIAL PRIMARY KEY,
+    staging_file_id BIGINT,
+    location TEXT NOT NULL,
+    source_id VARCHAR(64) NOT NULL,
+    destination_id VARCHAR(64) NOT NULL,
+    destination_type VARCHAR(64) NOT NULL,
+    table_name TEXT NOT NULL,
+    total_events BIGINT,
+    created_at TIMESTAMP NOT NULL);
+
+ALTER TABLE wh_load_files ALTER COLUMN table_name TYPE TEXT;
+
+ALTER TABLE wh_load_files ADD COLUMN IF NOT EXISTS total_events BIGINT;
+
+CREATE INDEX IF NOT EXISTS wh_load_files_source_destination_id_table_name_index ON wh_load_files (source_id, destination_id, table_name);
+
+--
+-- wh_uploads
+--
+
+DO $$ BEGIN
+    DROP TYPE wh_upload_state_type;
```

Comment:
This should be afrer `ALTER TABLE wh_uploads ALTER COLUMN status TYPE VARCHAR(255);` ?